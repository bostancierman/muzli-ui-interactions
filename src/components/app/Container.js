import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Immutable, { Map } from 'immutable'
import { map } from 'react-immutable-proptypes'
import { project } from 'src/selectors/project'
import { startTimer, stopTimer } from 'src/actions/project'
import { StyldH1, StyldH2 } from 'src/components/common/styled'
import Input from 'react-ions/lib/input'
import Button from 'react-ions/lib/button'
import style from 'src/styles/base.scss'

export class Container extends Component {
  constructor(props) {
    super(props)
  }

  static propTypes = {
    project: map
  }

  shouldComponentUpdate = () => {
    return false
  }

  render = () => {
    return (
      <div>
        <StyldH1 key='h1'>This is Boilerplate. ðŸ”¥</StyldH1>
        <StyldH2 key='h2'>Application state</StyldH2>
        <Input key='input' />
        <Button key='start' onClick={this.props.actions.startTimer}>Start</Button>
        <Button key='stop' onClick={this.props.actions.stopTimer}>Stop</Button>
        <div key='code'>
          <code>{JSON.stringify(this.props.project.toJS())}</code>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    project: project(state)
  }
}

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators({
      startTimer,
      stopTimer
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Container)
