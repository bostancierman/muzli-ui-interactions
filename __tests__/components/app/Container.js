import React from 'react'
import Immutable from 'immutable'
import { Container } from 'src/components/app/Container.js'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const props = {
    actions: {
      startTimer: () => {},
      stopTimer: () => {}
    },
    project: Immutable.fromJS({})
  }

  const tree = renderer.create(
    <Container project={props.project} actions={props.actions} />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})