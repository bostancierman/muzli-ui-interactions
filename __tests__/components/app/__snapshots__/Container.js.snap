// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="sc-bdVaJa ibTDPh"
  >
    This is Boilerplate. ðŸ”¥
  </h1>
  <h2
    className="sc-bwzfXH bGgvvP"
  >
    Application state
  </h2>
  <div
    className="input-component"
  >
    <div
      className="input-container"
      style={null}
    >
      <input
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyPress={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        style={Object {}}
        value=""
      />
    </div>
  </div>
  <button
    className="btn"
    disabled={undefined}
    onClick={[Function]}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    onMouseOut={undefined}
    onMouseOver={undefined}
    onMouseUp={undefined}
    style={undefined}
    type="button"
  >
    <em>
      Start
    </em>
  </button>
  <button
    className="btn"
    disabled={undefined}
    onClick={[Function]}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    onMouseOut={undefined}
    onMouseOver={undefined}
    onMouseUp={undefined}
    style={undefined}
    type="button"
  >
    <em>
      Stop
    </em>
  </button>
  <div>
    <code>
      {}
    </code>
  </div>
</div>
`;
